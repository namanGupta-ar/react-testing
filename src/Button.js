import React, { useState } from "react";

const Button = () => {
  const [data, setData] = useState("");
  return (
    <div>
      <p>Button</p>
      <button data-testid="btn-1" onClick={() => setData("updated data")}>
        Update Data
      </button>
      <button data-testid="btn-2" onClick={() => setData("")}>
        Empty Data
      </button>
      <h1 data-testid="btn-val">{data ? data : <strike>Empty</strike>}</h1>
      <p>Naman</p>
    </div>
  );
};

export default Button;
