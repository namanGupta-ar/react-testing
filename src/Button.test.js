import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer"
import Button from "./Button";

afterEach(() => {
    cleanup();
});

test("Update Button event testing", () => {
    render(<Button />);
    const btn1 = screen.getByTestId('btn-1');
    fireEvent.click(btn1);
    expect(screen.getByText("updated data")).toBeInTheDocument();
})

test("Empty Button event testing", () => {
    render(<Button />);
    const btn2 = screen.getByTestId('btn-2');
    const btnVal = screen.getByTestId('btn-val');
    expect(btnVal).toBeInTheDocument();
    fireEvent.click(btn2);
    expect(btnVal).toHaveTextContent('Empty');
})


test("matched snapshot", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
})