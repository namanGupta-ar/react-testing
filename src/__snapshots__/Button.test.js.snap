// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<div>
  <p>
    Button
  </p>
  <button
    data-testid="btn-1"
    onClick={[Function]}
  >
    Update Data
  </button>
  <button
    data-testid="btn-2"
    onClick={[Function]}
  >
    Empty Data
  </button>
  <h1
    data-testid="btn-val"
  >
    <strike>
      Empty
    </strike>
  </h1>
  <p>
    Naman
  </p>
</div>
`;
